### PR TITLE
Add a sympy 1.5 run for CircleCI

### PR DESCRIPTION
**Pull request by eric-wieser**
Relates to #94, although lets start with the last 2.7 release

### CODE REVIEW

**Comment by eric-wieser** [comment]
One of the failures for `examples/ipython/LaTeX.ipynb::Cell 10`, after some extra whitespace additions
```diff
  ion*} W
-     = (1/B) \left( X\cdot Y\right)  C \sqrt{\left( X\cdot Y\right) ^{2} - 2 \left( X\cdot Y\right)  \left( X\cdot e\right)  \left( Y\cdot e\right) }
-     =       \left( X\cdot Y\right)  C
-     - (1/B) \left( X\cdot e\right)  \left( Y\cdot e\right)  C \sqrt{\left( X\cdot Y\right) ^{2} - 2 \left( X\cdot Y\right)  \left( X\cdot e\right)  \left( Y\cdot e\right) }
+     -       \left( X\cdot e\right)  \left( Y\cdot e\right)  C
-     + (1/B) \left( X\cdot e\right)  \left( Y\cdot e\right)  \sqrt{\left( X\cdot Y\right) ^{2} - 2 \left( X\cdot Y\right)  \left( X\cdot e\right)  \left( Y\cdot e\right) }
+     +       \left( X\cdot e\right)  \left( Y\cdot e\right)
      + S \sqrt{\left( X\cdot Y\right) ^{2} - 2 \left( X\cdot Y\right)  \left( X\cdot e\right)  \left( Y\cdot e\right) } \end{equation*}
  \begin{equation*} \text{Scalar Coefficient}
-     = (1/B) \left( X\cdot e\right)  \left( Y\cdot e\right)  \sqrt{\left( X\cdot Y\right) ^{2} - 2 \left( X\cdot Y\right)  \left( X\cdot e\right)  \left( Y\cdot e\right) }
+     =       \left( X\cdot e\right)  \left( Y\cdot e\right) \end{equation*}
  \end{equation*}
  \begin{equation*} \text{Cosh Coefficient}
-     = (1/B) \left( X\cdot Y\right)                          \sqrt{\left( X\cdot Y\right) ^{2} - 2 \left( X\cdot Y\right)  \left( X\cdot e\right)  \left( Y\cdot e\right) }
-     - (1/B) \left( X\cdot e\right)  \left( Y\cdot e\right)  \sqrt{\left( X\cdot Y\right) ^{2} - 2 \left( X\cdot Y\right)  \left( X\cdot e\right)  \left( Y\cdot e\right) }
+     =       \left( X\cdot Y\right)  - \left( X\cdot e\right)  \left( Y\cdot e\right)
  \end{equation*}
```

**Comment by eric-wieser** [comment]
Down to just two failures

**Comment by eric-wieser** [comment]
One failure!

```
#x1B[91mNotebook cell execution failed#x1B[0m
#x1B[94mCell 2: Cell outputs differ

Input:
#x1B[0m!python eval_check.py

#x1B[94mTraceback:#x1B[0m
#x1B[94m mismatch 'stdout'#x1B[91m

 assert reference_output == test_output failed:

  'Frame = (\x1...ex))*eu)\n3\n' == 'Frame = (\x1...ex))*eu)\n3\n'
  Skipping 171 identical leading characters in diff, use -v to show
  - eu.ev_r = -1/2 + 1/2
  ?           -    ^
  + eu.ev_r = 1/2 - 1/2
  ?               ^
    ev.eu_r =  0 
    ev.ev_r = 1
    Frame = (#x1B[0;34mex#x1B[0m + #x1B[0;34mey#x1B[0m + #x1B[0;34mez#x1B[0m,#x1B[0;34mex#x1B[0m - #x1B[0;34mey#x1B[0m)
    Reciprocal Frame = (#x1B[0;34mex#x1B[0m/3 + #x1B[0;34mey#x1B[0m/3 + #x1B[0;34mez#x1B[0m/3,#x1B[0;34mex#x1B[0m/2 - #x1B[0;34mey#x1B[0m/2)
    eu.eu_r = 1
  - eu.ev_r = -1/2 + 1/2
  ?           -    ^
  + eu.ev_r = 1/2 - 1/2
  ?               ^
    ev.eu_r =  0 
    ev.ev_r = 1
    eu = #x1B[0;34mex#x1B[0m + #x1B[0;34mey#x1B[0m + #x1B[0;34mez#x1B[0m
    ev = #x1B[0;34mex#x1B[0m - #x1B[0;34mey#x1B[0m
    eu^ev|ex
    (eu^(ev|ex))
    #x1B[0;34mex#x1B[0m + #x1B[0;34mey#x1B[0m + #x1B[0;34mez#x1B[0m
    eu^ev|ex*eu
    ((eu^(ev|ex))*eu)
    3
#x1B[0m
```

File: test/.nbval_sanitize.cfg [hunk 1]
```diff
@@ -10,3 +10,29 @@ replace: $
...
+regex: \\begin\{matrix}- \\frac\{\\sqrt\{2}}\{4} \+ \\frac\{1}\{2}\\end\{matrix}
+replace: \\begin{matrix}\\frac{1}{2} - \\frac{\\sqrt{2}}{4}\\end{matrix}
+regex: =( +)- \\frac\{\\sqrt\{2}}\{4} \+ \\frac\{1}\{2}(?=\$| \\end\{equation\*})
+replace: =\1\\frac{1}{2} - \\frac{\\sqrt{2}}{4}
+
+regex: \{- \\left\( e_\{E}\\cdot e_\{B}\\right\) \^\{2} \+ 1\}
+replace: {1 - \\left( e_{E}\\cdot e_{B}\\right) ^{2}}
+
+regex: eu\.ev_r = -1/2 \+ 1/2(?=\n)
+replace: eu.ev_r = 1/2 - 1/2
```
Comment (utensil) [inline test/.nbval_sanitize.cfg]: Aha, you use this to make both version happy

Comment (utensil) [reply test/.nbval_sanitize.cfg]: We better use the 1.5 output as the default, and replace the old one to the new one here?

Comment (eric-wieser) [reply test/.nbval_sanitize.cfg]: We can do that in a follow-up

The sanitizations apply both to the expected and actual result - so will keep working when we regenerate the notebook with 1.5.0

File: .circleci/config.yml [hunk 2]
```diff
@@ -28,6 +32,13 @@ defaults: &defaults
         path: test-reports
 
 jobs:
+  "python-3.7-sympy-1.5":
+    <<: *defaults
+    environment:
+      PIP_EXTRA_INSTALLATION: sympy==1.5
```
[the remark this answers is not in the document]
Comment (eric-wieser) [reply .circleci/config.yml]: That was my intent here, yeah

**Pull request merged**